src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1541,35 +1541,52 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
             plan = parse_resource_id(plan)['name']
         plan_info = client.app_service_plans.get(resource_group_name, plan)
         if not plan_info:
             raise CLIError("The plan '{}' doesn't exist".format(plan))
         location = plan_info.location
-        is_linux = plan_info.reserved
-        if is_linux:
-            functionapp_def.kind = 'functionapp,linux'
-            site_config.app_settings.append(NameValuePair('FUNCTIONS_EXTENSION_VERSION', 'beta'))
-            site_config.app_settings.append(NameValuePair('MACHINEKEY_DecryptionKey',
-                                                          str(hexlify(urandom(32)).decode()).upper()))
-            if deployment_container_image_name:
-                site_config.app_settings.append(NameValuePair('DOCKER_CUSTOM_IMAGE_NAME',
-                                                              deployment_container_image_name))
-                site_config.app_settings.append(NameValuePair('FUNCTION_APP_EDIT_MODE', 'readOnly'))
-                site_config.app_settings.append(NameValuePair('WEBSITES_ENABLE_APP_SERVICE_STORAGE', 'false'))
-            else:
-                site_config.app_settings.append(NameValuePair('WEBSITES_ENABLE_APP_SERVICE_STORAGE', 'true'))
-                site_config.linux_fx_version = 'DOCKER|appsvc/azure-functions-runtime'
-        else:
-            functionapp_def.kind = 'functionapp'
-            site_config.app_settings.append(NameValuePair('FUNCTIONS_EXTENSION_VERSION', '~1'))
-
+        is_linux = plan_info.reserved if plan_info.reserved else is_linux
         functionapp_def.server_farm_id = plan
         functionapp_def.location = location
 
     con_string = _validate_and_get_connection_string(cmd.cli_ctx, resource_group_name, storage_account)
+    if is_linux:
+        functionapp_def.kind = 'functionapp,linux'
+        site_config.app_settings.append(NameValuePair('FUNCTIONS_EXTENSION_VERSION', '~2'))
+        if consumption_plan_location:
+            site_config.app_settings.append(NameValuePair('FUNCTIONS_WORKER_RUNTIME', runtime))
+            linux_mode = linux_mode.lower()
+            if linux_mode == 'dev':
+                site_config.app_settings.append(NameValuePair('WEBSITE_CONTENTAZUREFILECONNECTIONSTRING',
+                                                                con_string))
+                site_config.app_settings.append(NameValuePair('WEBSITE_CONTENTSHARE',
+                                                                'consumption_linux_cli_contentShare'))
+            else:  # prod is the default value
+                if zip_url is None:
+                    logger.warning('''No zip_url provided, all requests to the created function app
+                                    will result in 503s''')
+                else:
+                    site_config.app_settings.append(NameValuePair('WEBSITE_CONTENTAZUREFILECONNECTIONSTRING',
+                                                                    zip_url))
```

Comment:
This should go in `WEBSITE_USE_ZIP` **not** `WEBSITE_CONTENTAZUREFILECONNECTIONSTRING`